### PR TITLE
Assorted cleanups

### PR DESCRIPTION
**Pull request by keszybz**
A bunch of tweaks accumulated while working on other stuff.

### CODE REVIEW

**Review by bluca — APPROVED** [review]
LGTM - maybe rebase after https://github.com/systemd/systemd/pull/19455 is merged to get a clean run

File: src/basic/io-util.h [hunk 1]
```diff
@@ -33,22 +33,20 @@ static inline size_t IOVEC_TOTAL_SIZE(const struct iovec *i, size_t n) {
...
-static inline size_t IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
+static inline void IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
         size_t j;
 
         for (j = 0; j < n; j++) {
                 size_t sub;
 
                 if (_unlikely_(k <= 0))
-                        break;
+                        return;
```
Comment (poettering) [inline src/basic/io-util.h]: maybe make this an assert() then?

**Comment by keszybz** [comment]
I added the `assert` that @poettering requested, but I also ended up changing that patch to return something after all, because this allows one caller in write_to_syslog() to avoid another loop to calculate iovec array length. It is a hot path after all, so making it a bit better seems worthwhile.

File: src/basic/io-util.h [hunk 2]
```diff
@@ -33,22 +33,27 @@ static inline size_t IOVEC_TOTAL_SIZE(const struct iovec *i, size_t n) {
         return r;
 }
 
-static inline size_t IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
-        size_t j;
+static inline bool IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
+        /* Returns true if there is nothing else to send (bytes written cover all of the iovec),
+         * false if there's still work to do. */
```
Comment (poettering) [inline src/basic/io-util.h:38]: hmm, EOF is usually indicated by commands such as read() via a ret val of 0, i.e. false. i.e. wouldn't it make more sense to invert the return value here? i.e. return true if there's more, and false if there is not?

Comment (keszybz) [reply src/basic/io-util.h:38]: I think the analogy with read() is pretty weak: with read() we return how many bytes were available, and interpret 0 to mean that none were so we're done. Here, we are reporting the result of incrementing: did we manage to increment all the way to the end (return true), or did we fall short (return false). I can invert it if you still prefer, but I think the current approach makes sense.

**Comment by keszybz** [comment]
bionic-i386: test-process-util: timeout. Should be unrelated.

**Comment by keszybz** [comment]
s390x failed in "boot-and-services". Should be unrelated too.

File: src/basic/io-util.h (hunk 2 above)
Comment (keszybz) [reply src/basic/io-util.h:38]: I merged this without changing this. If you still think it should be changed, let's do that separately.

**Pull request merged**